### PR TITLE
Iphone and android chrome input fixes

### PR DESCRIPTION
**Pull request by simonluu**
Fixes issue #669 
Changed auto focus to false on ios since we don't want to auto focus most forms on mobile. Changing auto focus to false on android messes up the input, so auto focus is set to true and it works.

### CODE REVIEW

**Comment by williamstein** [comment]
@timothyclemansinsea or @haraldschilly -- can you review this now?   I don't have an android phone....

File: src/smc-webapp/editor_chat.cjsx [hunk 1]
```diff
@@ -1022,7 +1022,7 @@ ChatRoom = (name) -> rclass
                 <Row>
                     <Col xs={10} style={padding:'0px 2px 0px 2px'}>
                         <Input
-                            autoFocus   = {true}
+                            autoFocus   = {if isMobile.Android() then true else false}
```
Comment (haraldschilly) [inline src/smc-webapp/editor_chat.cjsx]: `isMobile.Android()` already returns a boolean value, so just write `isMobile.Android()` there

Comment (haraldschilly) [inline src/smc-webapp/editor_chat.cjsx]: I fixed this, don't worry about it

**Comment by haraldschilly** [comment]
my only trouble with reviewing this is, that my dev project for the webapp is still broken. that's so ridiculous :expressionless:

File: src/smc-webapp/editor_chat.cjsx (hunk 1 above)
Comment (haraldschilly) [inline src/smc-webapp/editor_chat.cjsx]: oh, and more general, maybe you both actually want it the other way around ? I'm trying to test it.

**Comment by haraldschilly** [comment]
Well, by experimenting around, I was able to more or less test and verify it. My observation is, that the input box is always focused on android. What I don't know is, what happens on mobile non-android systems? Is it still possible to enter anything? What if that `autoFocus=true` is only set if on Android and otherwise it isn't set at all, would that work on iOS mobile devices?

Besides that I am not able to really follow what's going on here any more. The associated ticket is set to be closed and working 9h ago, 4h later william asks to review this on android. Wouldn't it be less confusing, if the ticket is closed after the fix is deployed and verified to work?

My assumption is, that setting `autoFocus=false` on iOS mobile devices does work and that's why the original ticket is closed.

**Comment by simonluu** [comment]
Setting autoFocus=false on iOS is fine but setting autoFocus=false on android breaks the input and won't let you type.

**Comment by williamstein** [comment]
> Wouldn't it be less confusing, if the ticket is closed after the fix is deployed and verified to work?

The original ticket title is "On an iPhone using Google Chrome with SMC chatrooms, it is not possible to type input." and it is definitely the case (as verified by my iPhone) that on live SMC right now things work fine. 

From what you guys say above, it seems safe to merge this.

**Comment by williamstein** [comment]
WAIT: You say above "Setting autoFocus=false on iOS is fine"... except that right now in the live site the running code is `autoFocus   = {true}` and everything seems to work fine on iPhone.  The whole point of this PR seems to be to break iPhone support while maintaining android support?  Huh?

I think I should just close this PR.

**Comment by williamstein** [comment]
Hold on -- I didn't realize this was already merged.  Let me just say: whatever is live NOW works perfectly on iphone so I'm a happy camper.

**Comment by simonluu** [comment]
it doesn't matter if autoFocus is set to true or false on iOS, it works either way. Setting it to false is for issue #670 . I tried setting it to false for android but that just breaks the input for android.

**Comment by haraldschilly** [comment]
So, that's what did confuse me. The very first commit here says it solves the ticket #669 … and it got closed after that. Everything else is just code cleanup. The autoFocus=true is definitely necessary for android.

**Pull request merged**